### DISCUSSION TITLE
Scroll use mouse wheel is too quickly for me, is there possible to change this behavior?

### DISCUSSION THREAD

**Discussion by zw963** [wezterm/wezterm#4947] [Q&A] [answered]
### What Operating System(s) are you running on?

Linux Wayland

### Which Wayland compositor or X11 Window manager(s) are you using?

GNOME 45 + wayland

### WezTerm version

20240128-202157-1e552d76

### Ask your question!

The default behavior when scroll use mouse wheel is too quickly for me, what i expected is a more slowly, smoothly experiences, just like konsole.

I could not found the docs for this, there is no scrolling event list  on [default-mouse-assignments](https://wezfurlong.org/wezterm/config/mouse.html#default-mouse-assignments), but when i disable all mouse event use `config.disable_default_mouse_bindings = true`,  mouse wheel was disabled too, so i consider the document need to update?

Thanks

**Comment by stickyburn** [comment]
Just faced the same! For me it scrolls the page way too high in a single scroll up. I can see actions to set scroll amount, but no binding for mouse scroll. There's middle down but that's obviously not it.

**Comment by bgrasmo** [comment, marked as answer]
I just encountered this myself, but found out how to modify it:
```lua
local act = wezterm.action

-- Change mouse scroll amount
config.mouse_bindings = {
  {
    event = { Down = { streak = 1, button = { WheelUp = 1 } } },
    mods = 'NONE',
    action = act.ScrollByLine(-3),
  },
  {
    event = { Down = { streak = 1, button = { WheelDown = 1 } } },
    mods = 'NONE',
    action = act.ScrollByLine(3),
  },
}
```
This scrolls by 3 lines up and down, which it turns out was what my previous terminal was doing.

I found the `ScrollByLine` action in the [KeyAssignment](https://wezfurlong.org/wezterm/config/lua/keyassignment/index.html) part of the documentation. There are others there, like `ScrollByPage`, `ScrollToTop` and `ScrollToBottom` which are interesting as well.

**Reply by zw963** [reply]
Cool, i found the example code for `WheelUp`, `WheelDown` appear on the [document)(https://wezfurlong.org/wezterm/config/mouse.html#configuring-mouse-assignments) just now, i probably ignore those code before, And I don't know be can use mods as `NONE` before, thanks.

**Reply by chbybnwr** [reply]
this breaks scrolling in vim

**Reply by zw963** [reply]
> this breaks scrolling in vim

you can disable this settings for checking current running process is VIM. it is possible.

**Reply by RyanMillerC** [reply]
Add `alt_screen = false` to fix scrolling in vim (or other alternate buffers).

```lua
config.mouse_bindings = {
  -- Slower scroll up/down (3 lines instead of Page Up/Down)
  {
    event = { Down = { streak = 1, button = { WheelUp = 1 } } },
    mods = 'NONE',
    action = wezterm.action.ScrollByLine(-3),
    alt_screen = false,
  },
  {
    event = { Down = { streak = 1, button = { WheelDown = 1 } } },
    mods = 'NONE',
    action = wezterm.action.ScrollByLine(3),
    alt_screen = false,
  },
}
```

**Reply by Felixoid** [reply]
Oh my, I was looking for this solution for sooooo long.

For some reason, the Wheel behavior did change very sporadically with last updates, and I can't tweak it, because the vim did break.

Now, this part provides the best approach for me

```
config.mouse_bindings = {
  -- Scrolling with the mouse wheel
  {
    event = { Down = { streak = 1, button = { WheelUp = 1 } } },
    mods = 'NONE',
    action = act.ScrollByPage(-0.2),
    alt_screen = false,
  },
  {
    event = { Down = { streak = 1, button = { WheelDown = 1 } } },
    mods = 'NONE',
    action = act.ScrollByPage(0.2),
    alt_screen = false,
  },
}
```

**Reply by Dronakurl** [reply]
For me, neither of these approaches changes anything with the scroll behaviour. Not in VIM, not in a plain terminal. Nothing. If there is some advice how to debug this, I would be grateful!

**Comment by joshuataylor** [comment]
:thinking: This doesn't seem to work for me on MacOS.

The problem I have is that Wezterm (and some other apps, depending) have weird scroll issues as MacOS touchpad software is a bit janky sometimes.

Can I also disable left/right mouse scroll (this is a touchpad), as it seems a bit too sensitive with less, etc.

Normal wezterm scrollback touchpad scrolling up/down is fine (eg scrolling outside of a pager).

**Reply by Erik262** [reply]
I'm unable to scroll at all, except for the history, which does not make sense to me.

**Reply by joshuataylor** [reply]
On MacOS?

**Reply by Erik262** [reply]
Yes. When I scroll with the mouse, I am actually scrolling through the history of the commands I typed.
